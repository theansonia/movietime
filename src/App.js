import { useState } from 'react';
import './App.scss';
import Search from './components/Search';
import { Route, Switch, Link, Redirect} from "react-router-dom";
import TvSearch from './components/tvSearch';
import TrendingSearch from './components/TrendingSearch';
import Details from './components/Details';
import MovieDetails from './components/MovieDetails';


// export const LocationDisplay = () => {
//   const location = useLocation()

//   return <div data-testid="location-display">{location.pathname}</div>
// }

// function NoMatch() {
//   let location = useLocation();

//   return (
//     <div>
//       <h3>
//         No match
//       </h3>
//     </div>
//   );
// }

function App() {

  const [searchStatus, updateSearchStatus] = useState(false);
  const [category, updateCategory] = useState(null);

  return (

      <div id='app'>
        <header></header>
        <div id='navbar'>
        <Link to='/home' id="homebutton"><div  onClick={() => {updateSearchStatus(true); updateCategory(null)}}><svg id='homebtn'  width="48" height="48" viewBox="0 0 24 24" fill="" xmlns="http://www.w3.org/2000/svg">
                            <g  clipPath="url(#clip0)">
                            <rect width="24" height="24" fill="#212529"/>
                            <path id='homepath' fillRule="evenodd" clipRule="evenodd" d="M22 11.414V24H2V11.414L0.707 12.707L0 12L12 0L24 12L23.293 12.707L22 11.414ZM16 23H21V10.414L12 1.414L3 10.414V23H8V14H16V23ZM15 15.111H9V22.889H15V15.111Z" fill="#F8F9FA"/>
                            </g>
                            <defs>
                            <clipPath id="clip0">
                            <rect width="24" height="24" fill="white"/>
                            </clipPath>
                            </defs>
                            </svg>

        </div></Link>
        

            <Link id="filmbutton" to={`/movies`} className='option'  onClick={() => {updateCategory('Movie'); updateSearchStatus(true)}} ><div>Movies</div></Link>
            <Link id="tvbutton"  to={`/shows`} className='option' onClick={() => {updateCategory('TV'); updateSearchStatus(true)}} ><div>TV Shows</div></Link>

            
        </div>
        <Link to={'/home'}> 
        <div onClick={() => {updateCategory(null); updateSearchStatus(true)}} >
        <svg id="searchbtn" width="104" height="104" viewBox="0 0 104 104" fill="none" xmlns="http://www.w3.org/2000/svg">
            <g filter="url(#filter0_d)">
            <path id='searchpath' d="M100 48C100 74.5097 78.5097 96 52 96C25.4903 96 4 74.5097 4 48C4 21.4903 25.4903 0 52 0C78.5097 0 100 21.4903 100 48Z" fill="#343A40"/>
            <path id='searchpath' d="M100 48C100 74.5097 78.5097 96 52 96C25.4903 96 4 74.5097 4 48C4 21.4903 25.4903 0 52 0C78.5097 0 100 21.4903 100 48Z" fill="url(#paint0_linear)"/>
            </g>
            <path d="M61.7632 42.7662C61.7632 49.4941 55.7091 55.0325 48.1316 55.0325C40.5541 55.0325 34.5 49.4941 34.5 42.7662C34.5 36.0384 40.5541 30.5 48.1316 30.5C55.7091 30.5 61.7632 36.0384 61.7632 42.7662Z" stroke="#F8F9FA"/>
            <line y1="-0.5" x2="19.0436" y2="-0.5" transform="matrix(0.671291 0.741194 -0.804176 0.594391 57.2162 51.885)" stroke="#F8F9FA"/>
            <defs>
            <filter id="filter0_d" x="0" y="0" width="104" height="104" filterUnits="userSpaceOnUse" colorInterpolationFilters="sRGB">
            <feFlood floodOpacity="0" result="BackgroundImageFix"/>
            <feColorMatrix in="SourceAlpha" type="matrix" values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0"/>
            <feOffset dy="4"/>
            <feGaussianBlur stdDeviation="2"/>
            <feColorMatrix type="matrix" values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.25 0"/>
            <feBlend mode="normal" in2="BackgroundImageFix" result="effect1_dropShadow"/>
            <feBlend mode="normal" in="SourceGraphic" in2="effect1_dropShadow" result="shape"/>
            </filter>
            <linearGradient id="paint0_linear" x1="52" y1="0" x2="52" y2="96" gradientUnits="userSpaceOnUse">
            <stop stopColor="#6C757D"/>
            <stop offset="1" stopColor="white" stopOpacity="0"/>
            </linearGradient>
            </defs>
        </svg>



            </div></Link> 
    
            

        <Switch>
          <Route path='/movies' >
            <Search category={category} searchStatus={searchStatus} /> 
          </Route>
        </Switch>
        <Switch>
          <Route path='/shows' >
            <TvSearch category={category} searchStatus={searchStatus} /> 
          </Route>
        </Switch>
        <Switch>
          <Route path='/home' >
            <TrendingSearch category={category} searchStatus={searchStatus} /> 
          </Route>
        </Switch>        

        <Switch >
          <Route path={`/details/:name`} component={Details} />
        </Switch>
        

        <Switch >
          <Route path={`/moviedetails/:title`} component={MovieDetails} />

        </Switch>

        <Switch >
          <Route path="*">
            <Redirect to='/home'/>

            {/* Uncomment below for testing  */}
            {/* <NoMatch /> */}
          </Route>
        </Switch>
        
        <Route exact path="/">
          <Redirect to='/home' />
        </Route>    
 
        {/* <LocationDisplay /> */}

      </div>
 
    );
}

export default App;

