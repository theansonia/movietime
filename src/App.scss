@import url('https://fonts.googleapis.com/css2?family=Source+Sans+Pro:ital,wght@0,200;0,300;0,400;0,600;0,700;1,200;1,300;1,400;1,600;1,700;1,900&display=swap');

$font-stack: 'Source Sans Pro', sans-serif;
$background: #F8F9FA;
$primary-font: #212529;

// #root {
//   height: 100%;
//   width: 200vh;
//   overflow: scroll;

// }

html, body {
  height:100%;
  width:100%;
  margin:0;
  padding:0;
  font-family: $font-stack;
  overflow-x: hidden;
  overscroll-behavior-y: none;
  display: block;
  font-size: 1rem;
}

#app{
  width: 100%;
  display: flex;
  flex: 1 1 auto;
  min-height: 100vh;
  flex-direction: column;

}
.moviecontainer{
  display: flex;
  flex-direction: column;
  flex: 1 1 auto;
  min-height: 100%;
  gap: 1rem;
  height: 100%;
  align-items: center;
  justify-content: center;
  margin-top: 1rem;
  overflow: hidden;
  width: 100%;

}

#searchpath{
  &:hover{
    fill: darken(#6C757D, 15%);
  }
}
#navbar{
  background-color:#212529;
  height: 6rem;
  // width: 100%;
  display: flex; 
  justify-content: space-between;
  align-items: center;
  margin-bottom: 1.5rem;
  z-index: 14;
  padding-right: 1rem;
  padding-left: 1rem;

  @media (min-width: 600px) {
    height: 4rem;
  }
}

#trending{
  display: flex;
  align-items: center;
  justify-content: flex-start;
  flex-direction: center;
  border-radius: 1.4rem;
  max-width: 100%;
  margin-left: 1rem; 
  text-decoration: underline;
  font-weight: bold;
  margin-top: 1rem;


}


.option{
  font-size: 1.5rem;
  color: #F8F9FA;
  text-decoration: none;
  // margin-right: .5em;
    &:hover{
      text-decoration: underline;
      cursor: pointer;
      color: #ADB5BD;
    }
    @media (min-width: 600px) {
      font-size: 1.15rem;
    }
}
#x{
  z-index: 9999999999;
  // left: 75%;
  position: fixed;
  &:hover{
    cursor: pointer;
  }
}

#searchbtn{
  width: 3.5em;
  background-repeat: no-repeat;
  background-position: 50%;
  position: fixed;
  transition: top .2s;
  top: 80%;
  left: 42.25%;
  pointer-events: all;
  z-index: 18;
  overflow: visible;
  // -webkit-appearance: button;
 

  &:hover{
    cursor: pointer;
  }

}

form{
  width: 100%;
  // margin-top: 1rem;
  font-size: 2rem;
}

#searchbar{
  display: flex;
  justify-content: center;
  align-items: center;
  -webkit-text-size-adjust: 100%;
  background-color: darken(#6C757D, 15%);
  margin-bottom: 1rem;
  width: 80%;
  border-radius: 30px;

    &:after, :before{
      box-sizing: border-box;
    }
    
  
}

input{
  text-rendering: auto;
  letter-spacing: normal;
  word-spacing: normal;
  display: inline-block;
  text-align: start;
  appearance: auto;
  cursor: text;
  font-size: 2rem;
  caret-color: #F8F9FA;
  text-indent: .75rem;
    &:focus{
      outline-color: #ADB5BD;
    }
  
}

::placeholder{
  color: #F8F9FA;
  overflow: hidden;
  text-overflow: ellipsis;
  font-size: 1rem;
  text-indent: .4rem;
}

#search {
  position: relative;
  color: #F8F9FA;
  // font-size: 1rem;
  padding: 0 26px;
  // padding-left: 48px;
  background-color: darken(#6C757D, 15%);
  height: 4.125rem;
  line-height: 1.5rem;
  // border: none;
  height: 4rem;
  width: 100%;
  border-radius: 30px;
  font-size: 1.8rem;
  box-shadow: 0 10px 40px #212529, 0 0 0 20px #F8F9FA;
  
 

    &:after, :before{
    box-sizing: border-box;
    border-bottom: 2px solid #F8F9FA;
  }
}


#movies{
  min-width: 100%;
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  align-items: center;

}

#poster{
  align-items: center;
  width: 8em;
  justify-content: center;
  margin-left: .25rem;
  margin-right: .5rem;
}

#details{
  display: flex;
  font-size: .25rem;
  flex-direction: column;
  // padding-right: .65em;
  margin: 0;
  justify-content: flex-start;
  
}

#title{
  font-size: 1rem;
  font-weight: bold;
  margin-bottom: .1rem;
  // line-height: 2rem;
}

#release{
  color: darken(#ADB5BD, 5%);
  margin-bottom: .25em;


}

#overview{
  overflow: scroll;
  margin-bottom: .15em;
  text-align: left;
  --max-lines: 5;
  position: relative;
  max-height: calc(var(--lh) * var(--max-lines));
  overflow: hidden;
  padding-right: .1rem;
  display: -webkit-box;
  -webkit-line-clamp: 5;
  -webkit-box-orient: vertical; 
}

#rating{
  display: flex;
  justify-content: flex-start;
  margin-top: .25em;
}



.footer{
  margin-top: 1rem;
  padding: 1rem;
  background-color: none;
  position: fixed;
  z-index: 1;
  bottom: 0;
  left: 35%;
  width: 50%;
  display: flex;
  flex: 1;
}



#homebtn{
  width: 2em;
  margin-top: .3rem;
  // margin-right: 5em;
  // margin-right: 1.5em;
  &:hover{
    cursor: pointer;
  }
}

#homepath{
  &:hover{
    fill: darken(#ADB5BD, 18%);
  }
  
}
#more{
  color: #4cc9f0;
  text-decoration: underline;
  &:hover{
    cursor: pointer;
  }
}

#detail{
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

#detailposter{
  width: 14rem;
  margin-left: -.1rem;
  margin-top: .5rem;
}

#detaildetails{
  text-align: start;
  margin: .75em;
  text-decoration: none;
}

#detailtitle{
  font-size: 1.6em;
  margin-bottom: .5em;
  font-weight: bold;
  margin-top: .5em;
}

#detailoverview{
  margin-bottom: .5em;
}

#detailrelease{
  color: darken(#ADB5BD, 5%);
  margin-bottom: .5em;
}

#likethis{
  font-size: 1.2rem;
  font-weight: bold;
  text-align: left;
  align-self: flex-start;
  // margin-left: -10rem;
  text-decoration: underline;

}

#recs{
  
  display: flex;
  flex-direction: column;
  position: relative;
  // margin: 0 auto;
  align-items: center;
  justify-content: center;
  margin-left: .85rem;

  
}

#recposter{
  width: 95%;
  position: relative;
  margin: auto;
  border-radius: 3px;
  

}

#rectitle{
  color: white;
  width: 8.5rem;
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%, -50%);
  -ms-transform: translate(-50%, -50%);
  transform: translate(-50%, -50%);
  justify-content: center;
  text-align: center;
  // margin-top: .25em;
  // margin-bottom: .5em;
  font-size: 1.5rem;
    z-index: 1000000;
  
}

#type{
  text-decoration: underline;
  margin-bottom: .25rem;
  font-weight: 300;
}

#star{
  width: .5rem;
}


@media screen 
  and (min-device-width: 1200px) 
  and (max-device-width: 1600px) 
  and (-webkit-min-device-pixel-ratio: 2)
  and (min-resolution: 192dpi) {

  #star{
    width: .75rem;
  }
  // #recs{
  //   display: inline-grid;
  //   justify-content: center;
  // }
  
  #recposter{
    width: 14em;
    padding: .25rem;
    justify-items: center;
    border-radius: 3px;
    &:hover{
      box-shadow: 3px 3px 5px 6px #ccc;
    }
  }

  #rectitle{
    margin: 0;
    margin-top: .5em;
    font-size: .75rem;
    text-align: center;
    z-index: 1000000000;

    &:hover{
      z-index: 1000;
    }
  }

  .moviecontainer{
    width: 100%;
    margin: 0;
    margin-top: 1rem;
    
  }
  
  }
  #poster{
    width: 12em;
    margin-right: 1rem;
    margin-left: .5rem;

  }

  #title{
    font-size: 1.5rem;

  }

  #overview{
    overflow: scroll;
    margin-bottom: .15em;
    font-size: 3em;
    text-align: left;
    overflow: hidden;
    padding-right: .1rem;
    display: -webkit-box;
    -webkit-line-clamp: 8;
    -webkit-box-orient: vertical; 

  }

  #detailposter{
    width: 18rem;

  }
  #searchbtn{
    top: 80%;
    // left: 42.5%;

  }

  #searchbar{
    width: 80%;
    // margin-left: 2rem;
    margin: 0 auto;
    
    
  }
  
  input{
    font-size: 1.5rem;
    padding-left: 1rem;

  }
  
  ::placeholder{
    font-size: 1rem;
    padding-left: .5rem;
  }
  
  #x{
    left: 79%;
    // top: 1%;
  }

  #search {
    position: relative;
    color: #F8F9FA;
    // font-size: 1rem;
    padding: 0;
    // padding-left: 48px;
    height: 4.125rem;
    line-height: 1.5rem;
    height: 4rem;
    width: 100%;
    font-size: 1.6rem;
    border: none;
    // padding-left: .5rem;
 
  }



@media only screen 
  and (min-device-width: 770px) 
  and (max-device-width: 1024px) 
  and (-webkit-min-device-pixel-ratio: 1){

  #searchbtn{
    top: 77%;
    // left: 42.5%;
  }
      
  #star{
    width: .75rem;
  }
  
  #poster{
    width: 12rem;
  }

  #recposter{
    width: 24rem;
    padding: .25rem;
    border-radius: 3px;
    &:hover{
      box-shadow: 3px 3px 5px 6px #ccc;
    }
  }


  #rectitle{
    font-size: 3rem;
    width: 90%;
  }


  .moviecontainer{
    width: 100%;
    margin: 0;
    margin-top: 1rem;

  }

  #overview{
    font-size: 1.5rem;
    padding-right: .5rem;
    display: -webkit-box;
    -webkit-line-clamp: 7;
    -webkit-box-orient: vertical; 
  }

  #movies{
    width: 100%;
    display: flex;
    flex-direction: row;
    align-items: center;
  
  }

}
// tablet?
@media only screen 
  and (min-device-width: 800px) 
  // and (max-device-width: 480px)
  and (-webkit-min-device-pixel-ratio: 2)
  and (orientation: landscape) {

    
    #recs{
  
      display: inline-grid;
      justify-content: center;
      align-items: center;  
      margin: 0 auto;  
      
    }
    
    #recposter{
      width: 8.2rem;
      position: relative;
      border-radius: 3px;
      padding: .25rem;
      
    
    }
    #rectitle{
      width: 95%;
    }

  #searchbtn{
    left: 46.25%;
    top: 80%;
  }
 
  
}

// was meant for tablet but seems to be affecting laptop

@media only screen 
  and (min-device-width: 768px) 
  and (max-device-width: 1024px) 
  and (orientation: landscape) 
  and (-webkit-min-device-pixel-ratio: 1) {

    #recs{
  
      display: inline-grid;
      justify-content: center;
      align-items: center;    
      margin-left: 6.5rem;      
    }
    
    #recposter{
      width: 12rem;
      position: relative;
      margin: auto;
      border-radius: 3px;
      padding: .25rem;
    
    }

    #rectitle{
      font-size: 1rem;
      text-align: center;
    }
  #searchbtn{
    left: 45%;
    top: 80%;
  }
}

@media only screen 
  and (min-device-width: 375px) 
  and (max-device-width: 812px) 
  and (-webkit-min-device-pixel-ratio: 1)
  and (orientation: landscape) { 
    #recposter{
      width: 95%;
    }
    #rectitle{
      width: 28rem;
    }
}


@media only screen 
  and (min-device-width: 768px) 
  and (max-device-width: 1024px) 
  and (orientation: portrait) 
  and (-webkit-min-device-pixel-ratio: 1) {

    #recs{
  
      display: flex;
      // justify-content: center;
      // align-items: center;    
      
    }
    
    #recposter{
      width: 20rem;
      position: relative;
      margin: auto;
      border-radius: 3px;
      // padding: .25rem;
    
    }

    #rectitle{
      width: 16rem;
    }
  #searchbtn{
    left: 45%;
    top: 80%;
  }
}

// extra large screens
@media only screen
  and (min-width: 1600px)
  and (-webkit-min-device-pixel-ratio: 1) {
    #recs{
  
      display: inline-grid;
      justify-content: center;
      align-items: center;  
        
      @media only screen 
  and (min-device-width: 375px) 
  and (max-device-width: 812px) 
  and (-webkit-min-device-pixel-ratio: 3)
  and (orientation: portrait) { 

}
      
    }
    
    #recposter{
      width: 12.2rem;
      position: relative;
      margin: auto;
      border-radius: 3px;
    
    }

    #searchbtn{
      left: 48%;
      top: 90%;
    }

    #star{
      width: .75rem;
    }
    #overview{
      overflow: scroll;
      margin-bottom: .15em;
      text-align: left;
      --max-lines: 5;
      position: relative;
      max-height: calc(var(--lh) * var(--max-lines));
      overflow: hidden;
      padding-right: .1rem;
      display: -webkit-box;
      -webkit-line-clamp: 9;
      -webkit-box-orient: vertical; 
    }
    
  }