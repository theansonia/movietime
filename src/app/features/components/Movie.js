/* eslint-disable no-useless-escape */
import React from 'react'
import Stars from './Stars'
import { Link } from 'react-router-dom'
import { useDispatch } from 'react-redux'
import { changeCategory } from '../../appSlices/categorySlice'
import { setSearchStatus } from '../../appSlices/searchStatusSlice'

const Movie = ({
  title,
  poster,
  overview,
  release,
  rating,
  reviews,
  type,
  aired,
  id,
}) => {
  const dispatch = useDispatch()

  return (
    <div id='movies'>
      <Link
        onClick={() => {
          dispatch(setSearchStatus(false))
          dispatch(changeCategory(null))
        }}
        style={{ textDecoration: 'none' }}
        // eslint-disable-next-line no-useless-escape
        to={{
          pathname: `/moviedetails/${title.replace(
            /[.,/#!$%\^&\*;:{}=\-_`~()]/g,
            ''
          )}`,
          state: { id: id, type: type },
        }}
      >
        <div id='poster'>
          {poster !== 'https://image.tmdb.org/t/p/w500/null' &&
          poster !== 'https://image.tmdb.org/t/p/w500/undefined' ? (
            <img id='poster' src={poster} alt={`Movie poster for ${title}`} />
          ) : (
            <img
              id='poster'
              src='https://image.tmdb.org/t/p/w500//8SRUfRUi6x4O68n0VCbDNRa6iGL.jpg'
              alt='back up movie poster cinema pardiso'
            />
          )}
        </div>
      </Link>
      <div id='details'>
        <Link
          onClick={() => {
            dispatch(setSearchStatus(false))
            dispatch(changeCategory(null))
          }}
          style={{
            textDecoration: 'none',
            fontSize: '1em',
            marginBottom: '.5em',
            fontWeight: 'bold',
            marginTop: '.5em',
            color: 'inherit',
          }}
          to={{
            pathname: `/moviedetails/${title.replace(
              /[.,/#!$%\^&\*;:{}=\-_`~()]/g,
              ''
            )}`,
            state: { id: id, type: type },
          }}
        >
          {title ? <div id='title'>{title}</div> : null}
          <div id='type'>
            {type && type.length === 2 ? type.toUpperCase() : null}
          </div>
          <div id='type'>
            {type && type.length > 2
              ? type[0].toUpperCase() + type.substring(1)
              : null}
          </div>
        </Link>

        {release ? <div id='release'>Premiered {release}</div> : null}
        {aired ? <div id='release'>First Aired {aired}</div> : null}
        <Link
          onClick={() => {
            dispatch(setSearchStatus(false))
            dispatch(changeCategory(null))
          }}
          style={{
            textDecoration: 'none',
            fontSize: '1.4em',
            marginBottom: '.5em',
            fontWeight: 'normal',
            color: 'inherit',
          }}
          to={{
            pathname: `/moviedetails/${title.replace(
              /[.,/#!$%\^&\*;:{}=\-_`~()]/g,
              ''
            )}`,
            state: { id: id, type: type },
          }}
        >
          <div id='overview'>{overview}</div>
        </Link>

        {rating ? (
          <div id='rating'>
            {rating ? <Stars rating={rating} reviews={reviews} /> : 'No Rating'}
          </div>
        ) : null}
      </div>
    </div>
  )
}

export default Movie
