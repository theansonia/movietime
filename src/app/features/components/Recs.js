/* eslint-disable no-useless-escape */
import { useSelector } from 'react-redux'
import { Link } from 'react-router-dom'

const Recs = ({ title, pic, id }) => {
  const category = useSelector((state) => state.category)
  const topFunction = () => {
    console.log('hey')
    window.scrollTo(0, 0)
  }

  return (
    <div id='recs'>
      <Link
        style={{
          textDecoration: 'none',
          fontSize: '1em',
          marginBottom: '.5em',
          fontWeight: 'bold',
          marginTop: '.5em',
          color: 'inherit',
          zIndex: '100000000000',
        }}
        to={{
          pathname: `/moviedetails/${title.replace(
            /[.,/#!$%\^&\*;:{}=\-_`~()]/g,
            ''
          )}`,
          state: { id: id, category: category },
        }}
      >
        {title ? <div id='rectitle'>{title}</div> : null}
      </Link>
      <Link
        style={{ textDecoration: 'none' }}
        to={{
          pathname: `/moviedetails/${title.replace(
            /[.,/#!$%\^&\*;:{}=\-_`~()]/g,
            ''
          )}`,
          state: { id: id, category: category },
        }}
      >
        {pic !== 'https://image.tmdb.org/t/p/w500/null' ? (
          <div id='recposterdiv' onClick={topFunction}>
            <img
              id='recposter'
              style={{ borderRadius: '30px' }}
              src={pic}
              alt={`Movie poster for ${title}`}
            />
          </div>
        ) : (
          <div id='recposterdiv'>
            <img
              id='recposter'
              style={{ borderRadius: '30px' }}
              src='https://image.tmdb.org/t/p/w500///tfNuePdHrP9fp6K2VTJwHKKJf8C.jpg'
              alt='back up movie poster cinema pardiso'
            />
          </div>
        )}
      </Link>
    </div>
  )
}

export default Recs
