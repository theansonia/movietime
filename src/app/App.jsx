import React, { useState } from 'react'
import './App.scss'
import Search from './features/components/Search'
import { Route, Switch, Link, Redirect } from 'react-router-dom'
import TvSearch from './features/components/tvSearch'
import Details from './features/components/Details'
import MovieDetails from './features/components/MovieDetails'
import ThemeContextProvider from './contexts/ThemeContext'
import SearchButton from './features/components/SearchButton'
import { useDispatch, useSelector } from 'react-redux'
import changeCategory from '../actionCreators/changeCategory'
import ToggleTheme from './features/components/toggleTheme/ToggleTheme'
import TrendingSearch from './features/components/TrendingSearch'

function App() {
  const [searchStatus, updateSearchStatus] = useState(false)
  const [title, updateTitle] = useState('')
  const [pages, updatePages] = useState(1)
  const category = useSelector((state) => state.category)
  const dispatch = useDispatch()

  return (
    <ThemeContextProvider>
      {/* <PageContextProvider > */}
      <>
        {/* <header></header> */}
        <div id='navbar'>
          <Link
            to='/home'
            id='homebutton'
            onClick={() => {
              updateSearchStatus(false)
              updateTitle('')
              updatePages(1)
            }}
          >
            <div
              data-testid='homebtn'
              onClick={() => {
                updateSearchStatus(false)
                // dispatch(changeCategory(null))
                updatePages(1)
              }}
            >
              <svg
                id='homebtn'
                width='48'
                height='48'
                viewBox='0 0 24 24'
                fill=''
                xmlns='http://www.w3.org/2000/svg'
              >
                <g clipPath='url(#clip0)'>
                  <rect width='24' height='24' fill='#212529' />
                  <path
                    id='homepath'
                    fillRule='evenodd'
                    clipRule='evenodd'
                    d='M22 11.414V24H2V11.414L0.707 12.707L0 12L12 0L24 12L23.293 12.707L22 11.414ZM16 23H21V10.414L12 1.414L3 10.414V23H8V14H16V23ZM15 15.111H9V22.889H15V15.111Z'
                    fill='#F8F9FA'
                  />
                </g>
                <defs>
                  <clipPath id='clip0'>
                    <rect width='24' height='24' fill='white' />
                  </clipPath>
                </defs>
              </svg>
            </div>
          </Link>
          <Link
            id='filmbutton'
            to={'/movies'}
            className='option'
            onClick={() => {
              updateSearchStatus(false)
              dispatch(changeCategory('Movie'))
              updateTitle('')
              updatePages(1)
            }}
          >
            <div>Movies</div>
          </Link>
          <Link
            id='tvbutton'
            to={'/shows'}
            className='option'
            onClick={() => {
              updateSearchStatus(false)
              dispatch(changeCategory('TV'))
              updateTitle('')
              updatePages(1)
            }}
          >
            <div>TV Shows</div>
          </Link>
          <ToggleTheme />
        </div>

        <Switch>
          <Route path='/movies'>
            <Search
              category={category}
              searchStatus={searchStatus}
              title={title}
              updateTitle={updateTitle}
              updateSearchStatus={updateSearchStatus}
              updatePages={updatePages}
              pages={pages}
            />
          </Route>
          <Route path='/shows'>
            <TvSearch
              searchStatus={searchStatus}
              title={title}
              updateTitle={updateTitle}
              updateSearchStatus={updateSearchStatus}
              updatePages={updatePages}
              pages={pages}
            />
          </Route>
          <Route path='/home'>
            <TrendingSearch
              searchStatus={searchStatus}
              title={title}
              updateTitle={updateTitle}
              updateSearchStatus={updateSearchStatus}
              updatePages={updatePages}
              pages={pages}
            />
          </Route>
          <Route path={'/details/:name'} component={Details} />
          <Route path={'/moviedetails/:title'} component={MovieDetails} />
        </Switch>
        <Route exact path='/'>
          <Redirect to='/home' />
        </Route>

        <SearchButton
          searchStatus={searchStatus}
          updateSearchStatus={updateSearchStatus}
          updateTitle={updateTitle}
          updatePages={updatePages}
          pages={pages}
        />
      </>
      {/* </PageContextProvider> */}
    </ThemeContextProvider>
  )
}

export default App
