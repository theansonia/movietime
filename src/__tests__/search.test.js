import * as React from 'react'
import { render, fireEvent, cleanup } from '@testing-library/react'
import Search from '../components/Search'
import ReactDOM from 'react-dom'
import Details from '../components/Details'
import Recs from '../components/Recs'
import MovieDetails from '../components/MovieDetails'
import Movie from '../components/Movie'
import TVRecs from '../components/TVRecs'
import Show from '../components/Show'
import Stars from '../components/Stars'
import TrendingSearch from '../components/TrendingSearch'
import TvSearch from '../components/tvSearch'
import App from '../App'
import '@testing-library/jest-dom/extend-expect'

afterEach(cleanup)

xtest('change values via the fireEvent.change method', () => {
  const onChange = jest.fn()
  const { container } = render(
    <input type="text" data-testid="search" onChange={onChange} />
  )
  const input = container.firstChild
  fireEvent.change(input, { target: { value: 'a' } })
  expect(onChange).toHaveBeenCalledTimes(1)
  expect(input.value).toBe('a')
})

xtest('change value on input change', () => {
  const { container } = render(<Search />)
  expect(container.firstChild).toMatchSnapshot()
})

xtest('change value on input change trending search', () => {
  const { container } = render(<TrendingSearch />)
  expect(container.firstChild).toMatchSnapshot()
})

xtest('change value on input change tv search', () => {
  const { container } = render(<TvSearch />)
  expect(container.firstChild).toMatchSnapshot()
})

xtest('change value on input onchange', () => {
  const onChange = jest.fn()
  const { getByPlaceholderText } = render(<Search onChange={onChange} />)
  const searchInput = getByPlaceholderText(/Search for a Movie/i)
  expect(searchInput.value).toEqual('')
  // searchInput.onChange = onChange
  // fireEvent.change(searchInput, { target: { value: 'a' } });
  // expect(onChange).toHaveBeenCalledTimes(1)
})

xtest('change value on input onchange trending search', () => {
  const onChange = jest.fn()
  const { getByPlaceholderText } = render(
    <TrendingSearch onChange={onChange} />
  )
  const searchInput = getByPlaceholderText(/Search for Movies and TV Showss/i)
  expect(searchInput.value).toEqual('')
  // searchInput.onChange = onChange
  // fireEvent.change(searchInput, { target: { value: 'a' } });
  // expect(onChange).toHaveBeenCalledTimes(1)
})

xtest('change value on tvsearc onchange', () => {
  const onChange = jest.fn()
  const { getByPlaceholderText } = render(<TvSearch onChange={onChange} />)
  const searchInput = getByPlaceholderText(/Search for a TV Show/i)
  expect(searchInput.value).toEqual('')
  // searchInput.onChange = onChange
  // fireEvent.change(searchInput, { target: { value: 'a' } });
  // expect(onChange).toHaveBeenCalledTimes(1)
})

xdescribe('Search Component', () => {
  it('renders to DOM', () => {
    const div = document.createElement('div')
    ReactDOM.render(<Search />, div)
    ReactDOM.unmountComponentAtNode(div)
  })
})

xdescribe('Details Component', () => {
  it('renders to DOM', () => {
    const div = document.createElement('div')
    ReactDOM.render(<Details />, div)
    ReactDOM.unmountComponentAtNode(div)
  })
})

xdescribe('Recs Component', () => {
  it('renders to DOM', () => {
    const div = document.createElement('div')
    ReactDOM.render(<Recs />, div)
    ReactDOM.unmountComponentAtNode(div)
  })
})

xdescribe('MovieDetails Component', () => {
  it('renders to DOM', () => {
    const div = document.createElement('div')
    ReactDOM.render(<MovieDetails />, div)
    ReactDOM.unmountComponentAtNode(div)
  })
})

xdescribe('Movies Component', () => {
  it('renders to DOM', () => {
    const div = document.createElement('div')
    ReactDOM.render(<Movie />, div)
    ReactDOM.unmountComponentAtNode(div)
  })
})

xdescribe('TVrecs Component', () => {
  it('renders to DOM', () => {
    const div = document.createElement('div')
    ReactDOM.render(<TVRecs />, div)
    ReactDOM.unmountComponentAtNode(div)
  })
})

xdescribe('Show Component', () => {
  it('renders to DOM', () => {
    const div = document.createElement('div')
    ReactDOM.render(<Show />, div)
    ReactDOM.unmountComponentAtNode(div)
  })
})

// this test doesn't klike how blankstars array is filled
xdescribe('Stars Component', () => {
  it('renders to DOM', () => {
    const div = document.createElement('div')
    ReactDOM.render(<Stars />, div)
    ReactDOM.unmountComponentAtNode(div)
  })
})
