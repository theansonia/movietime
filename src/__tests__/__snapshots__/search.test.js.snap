// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`change value on input change 1`] = `
<div>
  <div
    id="searchbar"
  >
    <form>
      <input
        data-testid="search"
        id="search"
        list="movie"
        placeholder="Search for a Movie"
        value=""
      />
    </form>
  </div>
  <div
    id="trending"
  >
    Featured 
    s
  </div>
  <div
    class="moviecontainer"
  />
</div>
`;

exports[`change value on input change trending search 1`] = `
<div>
  <div
    id="searchbar"
  >
    <form>
      <input
        id="search"
        list="movie"
        placeholder="Search for Movies and TV Shows"
        value=""
      />
    </form>
  </div>
  <div
    id="trending"
  >
    Featured Movies and TV Shows
  </div>
  <div
    class="moviecontainer"
  />
</div>
`;

exports[`change value on input change tv search 1`] = `
<div>
  <div
    id="searchbar"
  >
    <form>
      <input
        id="search"
        list="movie"
        placeholder="Search for a TV Show"
        value=""
      />
    </form>
  </div>
  <div
    id="trending"
  >
    Featured TV Shows
  </div>
  <div
    class="moviecontainer"
  />
</div>
`;
